File: JAVA/AutoProject/KonzolosAuto.java
import java.util.Scanner;

public class KonzolosAuto implements Auto {
    private double sebesseg;

    public KonzolosAuto() {
        sebesseg = 0;
    }

    @Override
    public void gyorsul() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Mennyivel gyorsuljon a KonzolosAuto? ");
        double gyorsulas = scanner.nextDouble();
        sebesseg += gyorsulas;
    }

    @Override
    public double getSebesseg() {
        return sebesseg;
    }
}

File: C/AutoProject/konzolosauto.c
#include "konzolosauto.h"
#include <stdio.h>

void KonzolosAuto_Init(KonzolosAuto* self) {
    self->sebesseg = 0;
    self->base.gyorsul = KonzolosAuto_Gyorsul;
    self->base.getSebesseg = KonzolosAuto_GetSebesseg;
}

void KonzolosAuto_Gyorsul(Auto* self) {
    KonzolosAuto* konzolosAuto = (KonzolosAuto*)self;
    printf("Mennyivel gyorsuljon a KonzolosAuto? ");
    double gyorsulas;
    scanf("%lf", &gyorsulas);
    konzolosAuto->sebesseg += gyorsulas;
}

double KonzolosAuto_GetSebesseg(Auto* self) {
    KonzolosAuto* konzolosAuto = (KonzolosAuto*)self;
    return konzolosAuto->sebesseg;
}